docs/magento-layout-loader.md

```diff
@@ -0,0 +1,288 @@
+# magento-layout-loader
+
+_This is a very early implementation, and the API should be considered
+unstable._
+
+The `magento-layout-loader` is a [webpack
+loader](https://webpack.js.org/concepts/loaders/) that transforms
+[JSX](https://reactjs.org/docs/introducing-jsx.html) during compilation. It
+gives Magento modules/extensions the ability to inject or remove content blocks
+in a layout without modifying theme source files.
+
+## Terminology
+
+* **Container**: An HTML element that contains 0 or more ContainerChild
+  components. It acts as the target for the magento-loader-layout operations.
+* **ContainerChild**: Component exposed by
+  [Peregrine](https://github.com/magento-research/peregrine/). Responsible for
```

Comment:
I don't know if it's because the repository is private or not, but the link seems broken here.

Reply:
We're working on open-sourcing that repo